irohad/maintenance/metrics.cpp

```diff
@@ -4,93 +4,139 @@
-    // nevertheless dimensional values should only occur in low cardinality:
-    // https://prometheus.io/docs/practices/naming/#labels
-    auto& http_requests_counter = BuildCounter()
-        .Name("http_requests_total")
-        .Help("Number of HTTP requests")
-        .Register(*registry);
+  // ask the exposer_ to scrape the registry_ on incoming HTTP requests
+  exposer_->RegisterCollectable(registry_);
 
-    for (;;) {
-      std::this_thread::sleep_for(std::chrono::seconds(1));
-      const auto random_value = std::rand();
+  auto&block_height_gauge = BuildGauge()
+                                .Name("blocks_height")
+                                .Help("Total number of blocks in chain")
+                                //.Labels({{"label","a_metter"}})
+                                .Register(*registry_);
+  auto&block_height = block_height_gauge.Add({});//{{"value", "some"}});
+  block_height.Set(storage_->getBlockQuery()->getTopBlockHeight());
 
-      if (random_value & 1) tcp_rx_counter.Increment();
-      if (random_value & 2) tcp_tx_counter.Increment();
-      if (random_value & 4) udp_rx_counter.Increment();
-      if (random_value & 8) udp_tx_counter.Increment();
+  auto&peers_number_gauge = BuildGauge()
+      .Name("peers_number")
+      .Help("Total number peers to send transactions and request proposals")
+      //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&number_of_peers = peers_number_gauge.Add({});//{{"valueP", "any"}});
 
-      const std::array<std::string, 4> methods = {"GET", "PUT", "POST", "HEAD"};
-      auto method = methods.at(random_value % methods.size());
-      // dynamically calling Family<T>.Add() works but is slow and should be avoided
-      http_requests_counter.Add({{"method", method}}).Increment();
-    }
-  }).detach();
+  auto&domains_number_gauge = BuildGauge()
+      .Name("number_of_domains")
+      .Help("Total number of domains in WSV")
+      //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&domains_number = domains_number_gauge.Add({});
 
-  return {registry};
+  auto&total_number_of_transactions_gauge = BuildGauge()
+      .Name("number_of_signatures_in_last_block")
+      .Help("Number of signatures in last block")
+          //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&total_number_of_transactions = total_number_of_transactions_gauge.Add({});
+  
+  auto&number_of_signatures_in_last_block_gauge = BuildGauge()
+      .Name("number_of_signatures_in_last_block")
+      .Help("Number of signatures in last block")
+          //.Labels({{"label","a_metter"}})
+      .Register(*registry_);
+  auto&number_of_signatures_in_last_block = number_of_signatures_in_last_block_gauge.Add({});
+  
+  block_subscriber_ = std::make_shared<BlockSubscriber>(
+      getSubscription()->getEngine<EventTypes,BlockPtr>());
+  block_subscriber_->setCallback(
+        [&] //Values are stored in registry_
```

Comment:
It seems ```registry_``` passed here will be enough to update counters.

Reply:
done